flink-ml-lib/src/main/java/org/apache/flink/ml/common/param/HasL2.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.ml.common.param;
+
+import org.apache.flink.ml.param.DoubleParam;
+import org.apache.flink.ml.param.Param;
+import org.apache.flink.ml.param.ParamValidators;
+import org.apache.flink.ml.param.WithParams;
+
+/** Interface for the shared L2 regularization param. */
+public interface HasL2<T> extends WithParams<T> {
```

Comment:
Ideally we can make the parameter future proof.

According to the scikit-learn doc [1], it looks like the regularization can be one of `l1`, `l2` and `elasticnet`. And scikit-learn supports all choices.

Though Spark provides only the `HasElasticNetParam` without explicit `l1` or `l2` choices, the parameter doc suggests that `l1` or `l2` regularization is effectively used if user sets the parameter value to be `1` or `2`.

So both scikit-learn and Spark support all three modes. I guess we also want to be able to support these three modes in Flink ML, even if we support only one for now.

If we add `HasL2` here, how do we expect users to specify `L1` and `elasticnet` mode in the future? Should we use a double-valued `HasElasticNetParam` like Spark, or use a string-valued `HasPenalty` similar to Scikit-learn?

[1] https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LogisticRegression.html

[1] https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LogisticRegression.html

Reply:
Good observation! I prefer using `HasReg` and `HasElasticNet` to spefcify the three modes, similar as Spark.

I have renamed `HasL2` as HasReg`. Given that we are not using `HasElasticNet` in LogisticRegression, we can add this later.